docs/cli/classifier.py

```diff
@@ -0,0 +1,155 @@
+import os
+import sys
+import glob
+import re
+from pathlib import Path
+from collections import defaultdict
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+lib_path = os.path.abspath(os.path.join(current_dir, "..", ".."))
+sys.path.insert(0, lib_path)
+
+
+def classify_file_category(path):
+
+    relative_path = Path(path).relative_to(lib_path)
+    return_path = relative_path.as_posix()
+    filename = Path(*relative_path.parts[1:]).as_posix() if len(relative_path.parts) > 1 else return_path
+
+    if filename.startswith("linear"):
+        category = "linear"
+    elif filename.startswith("torch") or filename.startswith("nn"):
+        category = "nn"
+    else:
+        category = "general"
+    return category, return_path
+
+
+def fetch_option_flags(flags):
+    flag_list = []
+
+    for flag in flags:
+        flag_list.append(
+            {
+                "name": flag["name"].replace("\\", ""),
+                "instruction": flag["name"].split("-")[-1],
+                "description": flag["description"],
+            }
+        )
+
+    return flag_list
+
+
+def fetch_all_files():
+    main_files = [
+        os.path.join(lib_path, "main.py"),
+        os.path.join(lib_path, "linear_trainer.py"),
+        os.path.join(lib_path, "torch_trainer.py"),
+    ]
+    lib_files = glob.glob(os.path.join(lib_path, "libmultilabel/**/*.py"), recursive=True)
+    file_set = set(map(os.path.abspath, main_files + lib_files))
+    return file_set
+
+
+def find_config_usages_in_file(file_path, allowed_keys):
+    pattern = re.compile(r"\bconfig\.([a-zA-Z_][a-zA-Z0-9_]*)")
+    detailed_results = {}
+    try:
+        with open(file_path, "r", encoding="utf-8") as f:
+            lines = f.readlines()
+    except (IOError, UnicodeDecodeError):
+        return []
+
+    _, path = classify_file_category(file_path)
+
+    if file_path.endswith("main.py"):
+        for idx in range(len(lines)):
+            if lines[idx].startswith("def main("):
+                lines = lines[idx:]
+                main_start = idx
+                break
+        for i, line in enumerate(lines[1:]):
+            if line and line[0] not in (" ", "\t") and line.strip() != "":
+                lines = lines[:i]
+                break
+
+    for i, line in enumerate(lines, start=1):
+        matches = pattern.findall(line)
+        for key in matches:
+            if key in allowed_keys:
+                if key not in detailed_results:
+                    detailed_results[key] = {"file": path, "lines": []}
+                if file_path.endswith("main.py"):
+                    detailed_results[key]["lines"].append(str(i + main_start))
+                else:
+                    detailed_results[key]["lines"].append(str(i))
+
+    return detailed_results
```

Comment:
Similar in this function, 

```suggestion
def find_config_usages_in_file(file_path, allowed_keys, category_set):
    pattern = re.compile(r"\bconfig\.([a-zA-Z_][a-zA-Z0-9_]*)")
    try:
        with open(file_path, "r", encoding="utf-8") as f:
            lines = f.readlines()
    except (IOError, UnicodeDecodeError):
        return []

    # get start line in main.py
    if file_path.endswith("main.py"):
        for idx in range(len(lines)):
            if lines[idx].startswith("def main("):
                lines = lines[idx:]
                break

    all_str = " ".join(lines)
    matches = set(pattern.findall(all_str)) & allowed_keys
    
    category = classify_file_category(file_path)[0]
    for key in matches:
        category_set[category].add(key)
```

Reply:
TBD: try-catch here can be removed, as we want to see error instantly when someone change the files (not like production settings)